### ISSUE TITLE
Allow `npm_config_*`-style params for `no-process-env`

### ISSUE THREAD

**Issue by brettz9** [eslint/eslint#11808]
**What rule do you want to change?**

`no-process-env`

**Does this change cause the rule to produce more or fewer warnings?**

Fewer warnings.

**How will the change be implemented? (New option, new default behavior, etc.)?**

Either as a new option specifically for this form of variable, a configurable option allowing a user-customized pattern to indicate variables exempt from the rule, or a new default behavior specifically for this form of variable.

**Please provide some example code that this change will affect:**

```js
console.log(process.env.npm_config_test);
```

**What does the rule currently do for this code?**

Reports "Unexpected use of process.env"

**What will the rule do after it's changed?**

Report nothing.

**Are you willing to submit a pull request to implement this change?**

Not at this time.

**Rationale**

As per https://docs.npmjs.com/misc/config#environment-variables , npm has this `npm_config_*` means of passing in config through npm scripts. Although it is possible that environments could set these variables, it is more likely that npm scripts are, as designed, passing in this config.

**Comment by ljharb** [comment]
Note that these env vars aren’t guaranteed; they *set* npm config but I’m pretty sure npm doesn’t necessarily inject them. If you want to read npm config your best bet is to shell out to `npm config`.

**Comment by brettz9** [comment]
@ljharb : Sorry, I don't follow what the difference is. I am able to read `npm_config_*` vars from `process.env.npm_config_*` (e.g., such as `process.env.npm_config_test`) merely by passing in the argument to `npm`, e.g., `npm --test=abc test`.

This is a documented way to pass in config to scripts, and I'd expect a way to be able to make use of it in ESLint without complaint.

Admittedly, one can set through `npm config set` which makes it more like a normal environmental variable, but I think this is quite convenient to use in npm scripts and have the *option* to make an exception for these kinds of config.

I also neglected to mention `process.env.npm_package_config_`: https://docs.npmjs.com/misc/config#per-package-config-settings and https://docs.npmjs.com/files/package.json#config

**Comment by ljharb** [comment]
Fair enough. It seems pretty strange to rely on tho, when you can do `test=abc npm test`

**Comment by brettz9** [comment]
I like having a cross-platform way to express it--without needing [cross-env](https://github.com/kentcdodds/cross-env) or such.
